scipy/stats/_multivariate.py

```diff
@@ -5864,100 +5889,73 @@ class dirichlet_multinomial_gen(multi_rv_generic):
     def __init__(self, seed=None):
         super().__init__(seed)
         self.__doc__ = doccer.docformat(self.__doc__,
-                                        dirichlet_docdict_params)
+                                        dirichlet_mn_docdict_params)
 
-    def __call__(self, alpha, seed=None):
-        return dirichlet_multinomial_frozen(alpha, seed=seed)
+    def __call__(self, alpha, n, seed=None):
+        return dirichlet_multinomial_frozen(alpha, n, seed=seed)
 
-    def logpmf(self, alpha, x):
+    def logpmf(self, x, alpha, n):
         """The log of the probability mass function.
 
         Parameters
         ----------
-        %(_dirichlet_doc_default_callparams)s
         x: ndarray
-            A random vector of category counts distributed according to
-            a multinomial distribution.
+            A vector of category counts
+        %(_dirichlet_mn_doc_default_callparams)s
 
         Returns
         -------
-        out: scalar
+        out: ndarray or scalar
             Log of the probability mass function.
 
         """
 
-        n = np.sum(x, -1)
-
-        if alpha.shape != x.shape:
-            raise ValueError("`x` and `alpha` must have the same shape.")
-
-        if np.less(x, 0).any():
-            raise ValueError("`x` must not contain a non-negative integer.")
-
-        # Checks to make sure that every number in x can be expressed as an int
-        if not np.equal(np.mod(x, 1), 0).all():
-            raise ValueError("`x` must only contain integers.")
-
-        # Reduces line length
-        A = _dirichlet_multinomial_check_parameters(alpha)
-        B = np.sum(A, -1)
-        C = x + 1
-
-        out = loggamma(B) + loggamma(n + 1) - loggamma(n + B)
-        out += np.sum(loggamma(x + A) - (loggamma(A) + loggamma(C)), -1)
+        a, Sa, n, x = _dirichlet_multinomial_check_parameters(alpha, n, x)
```

Comment:
more recognizable variable names, yet more compact. For example, instead of `B`, `Sa` is not hard to read as "sum of a".